### PR TITLE
boot/bootstate16.go: clean snap_try_* vars when not in Trying status too

### PR DESCRIPTION
**Pull request by anonymouse64**
This is needed because the uc18/uc16 bootloader implementations do not actually do this, and instead snapd needs to reset the value.

Also modify the bootenv bootloader logic simulation to not clean this anymore, just as the real bootloaders do. 

This all requires some slight changes to the test logic, which since they are checking the BootVars map directly will see some additional empty keys in the map as markSuccessful will now write those empty keys. Eventually we should port these tests to use GetBootVars and not use the map directly since empty string keys show up the same as missing keys with GetBootVars and that would  have prevented needing to change these tests at all.

This will make it slightly simpler for the follow-up to #8340 which will refactor boottest/bootenv.go to use the bootloader logic simulation functions from boot_robustness_test.go uniformly. As such marking UC20 even though this change specifically is not UC20 related.

### CODE REVIEW

File: boot/bootstate16.go [hunk 1]
```diff
@@ -136,14 +136,18 @@ func (s16 *bootState16) markSuccessful(update bootStateUpdate) (bootStateUpdate,
 	env := u16.env
 	toCommit := u16.toCommit
 
+	tryBootVar := fmt.Sprintf("snap_try_%s", s16.varSuffix)
+	bootVar := fmt.Sprintf("snap_%s", s16.varSuffix)
+
 	// snap_mode goes from "" -> "try" -> "trying" -> ""
 	// so if we are not in "trying" mode, nothing to do here
 	if env["snap_mode"] != TryingStatus {
+		// clean the try var anyways in case it was leftover from a rollback,
```
Comment (pedronis) [inline boot/bootstate16.go:145]: does this needs a new unit test?

File: overlord/managers_test.go [hunk 2]
```diff
@@ -1843,6 +1844,7 @@ type: kernel`
 
 	c.Assert(bloader.BootVars, DeepEquals, map[string]string{
```
Comment (pedronis) [inline overlord/managers_test.go:1845]: we should probably make BootVars unexported at some point indeed, but not an immediate concern, so many other things

File: boot/bootstate16.go (hunk 1 above)
Comment (anonymouse64) [reply boot/bootstate16.go:145]: mmm probably yes I will try to write a couple small ones, it's a bit unfortunate that we have a significant delta between test coverage on uc16/uc18 boot logic than we do on uc20 now and it's a bit unclear how we could share a lot of the same code for that.

File: overlord/managers_test.go (hunk 2 above)
Comment (anonymouse64) [reply overlord/managers_test.go:1845]: yes agreed

File: boot/bootstate16.go (hunk 1 above)
Comment (anonymouse64) [reply boot/bootstate16.go:145]: some basic tests added now

**Pull request merged**